engine/src/main/java/org/hibernate/search/engine/backend/document/model/dsl/spi/IndexSchemaObjectNodeBuilder.java

```diff
@@ -27,6 +28,18 @@
 	<F> IndexSchemaFieldOptionsStep<?, IndexFieldReference<F>> addField(String relativeFieldName,
 			IndexFieldInclusion inclusion, IndexFieldType<F> indexFieldType);
 
+	/**
+	 * Create a new object filter and add it to the current builder.
+	 *
+	 * @param relativeNamedPredicateName The relative name of the new named predicate definition
+	 * @param inclusion Whether fields matching this template should be included, provided their parent is included.
+	 * @param factory The factory of the new object named predicate definition
+	 * @param <F> The type of named predicate factory for the new named predicate definition
+	 * @return A DSL step where the named predicate can be defined in more details.
+	 */
+	<F> IndexSchemaNamedPredicateOptionsStep<?> addNamedPredicate(String relativeNamedPredicateName,
```

Comment:
Type parameter `<F>` can be removed here and replaced with `NamedPredicateFactory` everywhere.